src/apps/angle/ethereum/angle.vaults.contract-position-fetcher.ts

```diff
@@ -12,7 +12,12 @@ import { Network } from '~types/network.interface';
 
 import { ANGLE_DEFINITION } from '../angle.definition';
 import { AngleContractFactory } from '../contracts';
-import { AngleApiHelper } from '../helpers/angle.api';
+
+const vaultAddresses = [
```

Comment:
i dont think this is a good idea, as vault managers could change frequently (new ones could be added), and this is going to be deployed on many chains.
I was about to open a PR for other chains.
So it will be a pain to change addresses all the time. I think it makes way more sense to just fetch the adresses

Reply:
ahh ok ok, that makes sense. How do you guys keep track of vault manager addresses ? Is there a way we could get a list calling a specific contract ? If not it's fine, i'll revert this part and keep the remaining stuff if it's fine with you.

Reply:
yes we could actually. using our Treasury there is a way to get all the vault managers. The problem is you dont know how many vaultmanagers there are so you need to loop over it. I dont know if it's ideal...
I'll write here a piece of code (probably a bit later tonight) to show you how we can do it, and you tell me if you prefer to integrate it like this.

Reply:
sounds good, thanks ! I'll have a look at both options and then let you know which one is more aligned with ours vision